docs/ref/settings.txt

```diff
@@ -403,6 +403,13 @@ Storing the CSRF token in a cookie (Django's default) is safe, but storing it
 in the session is common practice in other web frameworks and therefore
 sometimes demanded by security auditors.
 
+.. note::
+
+    Since the default error views use the CSRF token,
+    :class:`~django.contrib.sessions.middleware.SessionMiddleware` must appear
+    in :setting:`MIDDLEWARE` before any middleware that may raise an exception
```

Comment:
"raise an exception to trigger an error view (such as `PermissionDenied`)" may offer a bit more explanation.